From 2df2fe7f42803303e3d15dd0340cb9f39fc3d918 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:31:56 -0400
Subject: [iommu] vt-d: Use dev_is_pci() to check whether it is pci device

Message-id: <20140905013156.30867.18918.stgit@gir.stowe>
Patchwork-id: 91272
O-Subject: [RHEL7.1 PATCH 08/28] iommu/vt-d: Use dev_is_pci() to check whether it is pci device
Bugzilla: 1129808
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129808
Upstream Status: dbad086433af81513c84678070522455fefebe2a

commit dbad086433af81513c84678070522455fefebe2a
Author: Yijing Wang <wangyijing@huawei.com>
Date:   Thu Dec 5 19:43:42 2013 +0800

    iommu/vt-d: Use dev_is_pci() to check whether it is pci device

    Use PCI standard marco dev_is_pci() instead of directly compare
    pci_bus_type to check whether it is pci device.

    Signed-off-by: Yijing Wang <wangyijing@huawei.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 59779e1..04fbb88 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -2758,7 +2758,7 @@ static int iommu_no_mapping(struct device *dev)
  struct pci_dev *pdev;
  int found;
 
- if (unlikely(dev->bus != &pci_bus_type))
+ if (unlikely(!dev_is_pci(dev)))
   return 1;
 
  pdev = to_pci_dev(dev);
-- 
1.7.1